corehq/apps/users/static/users/js/web_users.js

```diff
@@ -73,6 +76,125 @@ hqDefine("users/js/web_users",[
+            expirationDate.setDate(expirationDate.getDate() + 31);
+            return (expirationDate - new Date()) / (24 * 60 * 60 * 1000);
+        });
+        self.isExpired = ko.computed(function () {
+            return self.daysRemaining() < 0;
+        });
+        self.daysRemainingText = ko.computed(function () {
+            return _.template(gettext("<%= days %> days remaining"))({
+                days: Math.floor(self.daysRemaining()),
+            });
+        });
+
+        self.actionMessage = ko.observable('');
+        self.actionInProgress = ko.observable(false);
+
+        self.visible = ko.observable(true);
+        self.remove = function () {
+            self.actionInProgress(true);
+            $.ajax(initialPageData.reverse("delete_invitation"), {
+                method: 'POST',
+                dataType: 'json',
+                data: {
+                    uuid: self.uuid,
+                },
+                success: function () {
+                    self.actionInProgress(false);
+                    self.visible(false);
+                },
+                error: function () {
+                    self.actionInProgress(false);
+                    self.actionMessage(gettext("Unable to delete invitation, please try again later."));
+                },
+            });
+        };
+
+        self.resend = function () {
+            self.actionInProgress(true);
+            $.ajax(initialPageData.reverse("reinvite_web_user"), {
+                method: 'POST',
+                dataType: 'json',
+                data: {
+                    uuid: self.uuid,
+                },
+                success: function (data) {
+                    self.actionInProgress(false);
+                    self.actionMessage(data.response);
+                    self.invited_on(new Date());
+                },
+                error: function () {
+                    self.actionInProgress(false);
+                    self.actionMessage(gettext("Unable to resend invitation, please try again later."));
+                },
+            });
+        };
+
+        return self;
+    };
+
+    var invitationsList = function (invitations) {
+        var self = {};
+        self.allInvitations = ko.observableArray(_.map(invitations, Invitation));
+        self.currentPageInvitations = ko.observableArray();
+
+        self.invitationToRemove = ko.observable();
+        self.confirmRemoveInvitation = function (model) {
+            self.invitationToRemove(model);
+        };
+        self.removeInvitation = function () {
+            self.invitationToRemove().remove();
+            self.invitationToRemove(null);
+        };
+        _.each(self.allInvitations(), function (i) {
+            // Invitations can be deleted, but not added back
+            i.visible.subscribe(function (newValue) {
+                if (!newValue) {
+                    self.allInvitations.remove(i);
+                    self.totalItems(self.totalItems() - 1);
```

Comment:
Is there a reason you didn't use a `computed` for `totalItems`? I can't remember if changing an observableArray triggers the computations.

Reply:
I think it would work either way - whatever code I was using as a reference just happened to use a regular observable instead of a computed one.